### ISSUE TITLE
 Add .tap() on promise

### ISSUE THREAD

**Issue by dudeman** [lhorie/mithril.js#228]
For chaining side effect functions on promises where you don't want to capture the return value from the function
```
m.request({...}).then(transform).tap(m.redraw).tap(log)
```
This was possible with .then in versions before 0.1.21
Changed with PR https://github.com/lhorie/mithril.js/pull/197

**Comment by lhorie** [comment]
Also should have `catch` (see #404)

**Comment by adriengibrat** [comment]
with ES6 promise API there is ' finally', the exemple above could be written like this:

```
m.request({...}).then(transform).finally(m.redraw).catch(log)
```
finally is always called (either on fullfill and reject) and return the same promise.

**Comment by isiahmeadows** [comment]
Closing, as this was fixed in #734.
